socketmode/socketmode_eventhandler.go

```diff
@@ -0,0 +1,197 @@
+package socketmode
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+)
+
+type SocketmodeHandler struct {
+	Client *Client
+
+	EventMap                       map[EventType][]SocketmodeHandlerFunc
+	InteractionEventMap            map[slack.InteractionType][]SocketmodeHandlerFunc
+	InteractionBlockActionEventMap map[string][]SocketmodeHandlerFunc
+	EventApiMap                    map[slackevents.EventAPIType][]SocketmodeHandlerFunc
+
+	Default SocketmodeHandlerFunc
+}
+
+// Handler have access to the event and socketmode client
+type SocketmodeHandlerFunc func(*Event, *Client)
+
+// Middleware accept SocketmodeHandlerFunc, and return SocketmodeHandlerFunc
+type SocketmodeMiddlewareFunc func(SocketmodeHandlerFunc) SocketmodeHandlerFunc
+
+func NewsSocketmodeHandler(client *Client) *SocketmodeHandler {
+	eventMap := make(map[EventType][]SocketmodeHandlerFunc)
+	interactionEventMap := make(map[slack.InteractionType][]SocketmodeHandlerFunc)
+	interactionBlockActionEventMap := make(map[string][]SocketmodeHandlerFunc)
+	eventApiMap := make(map[slackevents.EventAPIType][]SocketmodeHandlerFunc)
+
+	return &SocketmodeHandler{
+		Client:                         client,
+		EventMap:                       eventMap,
+		EventApiMap:                    eventApiMap,
+		InteractionEventMap:            interactionEventMap,
+		InteractionBlockActionEventMap: interactionBlockActionEventMap,
+		Default: func(e *Event, c *Client) {
+			log.Printf("Unexpected event type received: %v\n", e.Type)
```

Comment:
Prefer to use client's logger.

```suggestion
			c.log.Printf("unexpected event type received: %v\n", e.Type)
```